plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -0,0 +1,524 @@
+            table.insert(style_props, string.format("height: %spx", cur_image.height))
+        end
+        local style = table.concat(style_props, "; ")
+        return string.format([[<img src="%s" style="%s" alt=""/>]], cur_image.imgpath, style)
+    end
+    html = html:gsub("(<%s*img [^>]*>)", processImg)
+    logger.dbg("Images found in html:", images)
+
+    -- See what to do with images
+    local use_img_2x = false
+    if not include_images then
+        -- Remove img tags to avoid little blank squares of missing images
+        html = html:gsub("<%s*img [^>]*>", "")
+        -- We could remove the whole image container <div class="thumb"...> ,
+        -- but it's a lot of nested <div> and not easy to do.
+        -- So the user will see the image legends and know a bit about
+        -- the images he chose to not get.
+    end
+
+    UI:info(_("Building EPUB…"))
+    -- Open the zip file (with .tmp for now, as crengine may still
+    -- have a handle to the final epub_path, and we don't want to
+    -- delete a good one if we fail/cancel later)
+    local epub_path_tmp = epub_path .. ".tmp"
+    local ZipWriter = require("ffi/zipwriter")
+    local epub = ZipWriter:new{}
+    if not epub:open(epub_path_tmp) then
+        logger.dbg("Failed to open epub_path_tmp")
+        return false
+    end
+
+    -- We now create and add all the required epub files
+
+    -- ----------------------------------------------------------------
+    -- /mimetype : always "application/epub+zip"
+    epub:add("mimetype", "application/epub+zip")
+
+    -- ----------------------------------------------------------------
+    -- /META-INF/container.xml : always the same content
+    epub:add("META-INF/container.xml", [[
+<?xml version="1.0"?>
+<container version="1.0" xmlns="urn:oasis:names:tc:opendocument:xmlns:container">
+  <rootfiles>
+    <rootfile full-path="OEBPS/content.opf" media-type="application/oebps-package+xml"/>
+  </rootfiles>
+</container>]])
+    logger.dbg("Added META-INF/container.xml")
+
+    -- ----------------------------------------------------------------
+    -- OEBPS/content.opf : metadata + list of other files (paths relative to OEBPS/ directory)
+    -- Other possible items in this file that are of no interest to crengine :
+    --   In <manifest> :
+    --     <item id="cover" href="title.html" media-type="application/xhtml+xml"/>
+    --     <item id="cover-image" href="images/cover.png" media-type="image/png"/>
+    -- (crengine only uses <meta name="cover" content="cover-image" /> to get the cover image)
+    --   In <spine toc="ncx"> :
+    --     <itemref idref="cover" linear="no"/>
+    --   And a <guide> section :
+    --     <guide>
+    --       <reference href="title.html" type="cover" title="Cover"/>
+    --       <reference href="toc.html" type="toc" title="Table of Contents" href="toc.html" />
+    --     </guide>
+    local koreader_version = "KOReader"
+    if lfs.attributes("git-rev", "mode") == "file" then
```

Comment:
Could you use Version:getCurrentRevision() instead? Same thing, but only one place to change things.

https://github.com/koreader/koreader/blob/96179670f8d1d29f343db1ab0ce20ef6bd45acb2/frontend/version.lua#L7-L22